src/NServiceBus.Persistence.AzureStorage/SagaPersisters/AzureSagaPersister.cs

```diff
@@ -101,6 +99,29 @@ public Task Update(IContainSagaData sagaData, SynchronizedStorageSession session
                     }
                 }
             }
+
+            await RemoveSecondaryIndex(sagaData, table).ConfigureAwait(false);
+        }
+
+        static async Task RemoveSecondaryIndex(IContainSagaData sagaData, CloudTable table)
+        {
+            PartitionRowKeyTuple secondaryIndexKey;
+            if (secondaryIndexKeys.TryGetValue(sagaData, out secondaryIndexKey))
+            {
+                var e = new TableEntity
+                {
+                    ETag = "*"
+                };
+                secondaryIndexKey.Apply(e);
+                try
+                {
+                    await table.ExecuteAsync(TableOperation.Delete(e)).ConfigureAwait(false);
+                }
+                catch
+                {
+                    // this is only secondary index removal. If it fails, an oprhaned entity is the only cost
```

Comment:
>  an oprhaned entity is the only cost

Customers will be asking why we don't clean up (how this issue was raised for example).